scripts/install_third_party_test.py

```diff
@@ -0,0 +1,376 @@
+        with exists_swap, self.dir_exists_swap, self.url_retrieve_swap:
+            with self.remove_swap, self.rename_swap, self.extract_swap:
+                with url_open_swap, string_io_swap:
+                    install_third_party.download_and_unzip_files(
+                        'source url', 'target dir', 'zip root', 'target root')
+        self.assertEqual(
+            self.check_function_calls, self.expected_check_function_calls)
+        self.assertEqual(exists_arr, [False, True])
+
+    def test_download_and_untar_files(self):
+        exists_arr = []
+        def mock_exists(unused_path):
+            exists_arr.append(False)
+            return False
+        def mock_extractall(unused_self, unused_path):
+            self.check_function_calls['extractall_is_called'] = True
+
+        exists_swap = self.swap(os.path, 'exists', mock_exists)
+        extract_swap = self.swap(
+            tarfile.TarFile, 'extractall', mock_extractall)
+        unzip_swap = self.swap(
+            install_third_party, 'TMP_UNZIP_PATH', os.path.join(
+                RELEASE_TEST_DIR, 'tmp_unzip.tar.gz'))
+
+        with exists_swap, self.dir_exists_swap, self.url_retrieve_swap:
+            with self.remove_swap, self.rename_swap, unzip_swap, extract_swap:
+                install_third_party.download_and_untar_files(
+                    'source url', 'target dir', 'zip root', 'target root')
+        self.assertEqual(
+            self.check_function_calls, self.expected_check_function_calls)
+        self.assertEqual(exists_arr, [False])
+
+    def test_get_file_contents(self):
+        temp_file = tempfile.NamedTemporaryFile().name
+        actual_text = 'Testing install third party file.'
+        with python_utils.open_file(temp_file, 'w') as f:
+            f.write(actual_text)
+        self.assertEqual(
+            install_third_party.get_file_contents(temp_file), actual_text)
+
+    def test_return_json(self):
+        temp_file = tempfile.NamedTemporaryFile().name
+        actual_text = '{"Testing": "install_third_party"}'
+        with python_utils.open_file(temp_file, 'w') as f:
+            f.write(actual_text)
+        self.assertEqual(
+            install_third_party.return_json(temp_file),
+            {'Testing': 'install_third_party'})
+
+    def test_manifest_syntax_testing_with_valid_syntax(self):
+        install_third_party.test_manifest_syntax(
+            'files', {
+                'url': 'https://github.com/yui/yuicompressor/v2.4.8#md5=123456',
+                'files': ['yuicompressor-2.4.8.jar'],
+                'version': '2.4.8',
+                'targetDirPrefix': 'yuicompressor-',
+                'downloadFormat': 'files'})
+
+    def test_manifest_syntax_with_missing_mandatory_key(self):
+        print_arr = []
+        def mock_print(msg):
+            print_arr.append(msg)
+        print_swap = self.swap(python_utils, 'PRINT', mock_print)
```

Comment:
Once we migrate to py3 and remove python_utils, how will this work?

Reply:
We can [mock the inbuilt print function or stdout](https://stackoverflow.com/questions/12998908/mock-pythons-built-in-print-function) in that case.

Reply:
That's great!